search/task_helpers.py

```diff
@@ -0,0 +1,139 @@
+    """
+    def api_indexing_listener_inner(func):  # pylint: disable=missing-docstring
+        @wraps(func)
+        def wrapped_api_func(*args, **kwargs):  # pylint: disable=missing-docstring
+            reddit_obj = func(*args, **kwargs)
+            try:
+                indexing_func(reddit_obj)
+            except Exception:  # pylint: disable=broad-except
+                log.exception('Error occurred while trying to index PRAW object')
+            return reddit_obj
+        return wrapped_api_func
+    return api_indexing_listener_inner
+
+
+def index_full_post(post_obj):
+    """
+    Serializes a post object and runs a task to create an ES document for it.
+
+    Args:
+        post_obj (praw.models.reddit.submission.Submission): A PRAW post ('submission') object
+    """
+    data = {
+        'object_type': POST_TYPE,
+        'author': post_obj.author.name,
+        'channel_title': post_obj.subreddit.display_name,
+        'text': post_obj.selftext,
+        'score': post_obj.score,
+        'created': post_obj.created,
+        'post_id': post_obj.id,
+        'post_title': post_obj.title,
+        'num_comments': post_obj.num_comments,
+    }
+    return create_document.delay(
+        gen_post_id(post_obj.id),
+        data
+    )
+
+
+def update_post_text(post_obj):
+    """
+    Serializes post object text and runs a task to update the text for the associated ES document.
+
+    Args:
+        post_obj (praw.models.reddit.submission.Submission): A PRAW post ('submission') object
+    """
+    return update_document_with_partial.delay(
+        gen_post_id(post_obj.id),
+        {'text': post_obj.selftext}
+    )
+
+
+def update_post_removal_status(post_obj):
+    """
+    Serializes the removal status for a post object and runs a task to update that status
+    for the associated ES document.
+
+    Args:
+        post_obj (praw.models.reddit.submission.Submission): A PRAW post ('submission') object
+    """
+    return update_document_with_partial.delay(
+        gen_post_id(post_obj.id),
+        {'removed': bool(post_obj.banned_by) and not post_obj.approved_by}
+    )
+
+
+def _update_post_comment_count(comment_obj, incr_amount=1):
+    """
+    Updates the comment count for a post object (retrieved via a comment object)
+    and runs a task to update that count for the associated ES document.
+
+    Args:
+        comment_obj (praw.models.reddit.comment.Comment): A PRAW comment object
+        incr_amount (int): The amount to increment to count
+    """
+    post_obj = comment_obj.submission
+    return increment_document_integer_field.delay(
+        gen_post_id(post_obj.id),
+        field_name='num_comments',
+        incr_amount=incr_amount
+    )
+
+
+increment_post_comment_count = partial(_update_post_comment_count, incr_amount=1)
+decrement_post_comment_count = partial(_update_post_comment_count, incr_amount=-1)
+
+
+def update_indexed_score(instance, instance_type, vote_action=None):
```

Comment:
`vote_action` always seems required to me, so I'm not certain it should have a default value